app/src/main/java/com/example/frogweather/data/NetworkRepository.kt

```diff
@@ -3,11 +3,27 @@ package com.example.frogweather.data
 import java.io.IOException
 
 class NetworkRepository {
-    suspend fun getForecast(latitude: Double, longitude: Double, millis: Long) =
-        safeApiCall(call = { getForecastCall(latitude, longitude, millis) }, errorMessage = "Exception occurred")
+    suspend fun getDailyForecast(latitude: Double, longitude: Double, millis: Long) =
+        safeApiCall(call = { getDailyForecastCall(latitude, longitude, millis) }, errorMessage = "Exception occurred")
 
-    private suspend fun getForecastCall(latitude: Double, longitude: Double, millis: Long): CallResult<Forecast> {
-        val response = FrogWeatherApi.retrofitService.getForecast(latitude, longitude, millis)
+    suspend fun getHourlyForecast(latitude: Double, longitude: Double) =
+        safeApiCall(call = { getHourlyForecastCall(latitude, longitude) }, errorMessage = "Exception occurred")
+
+    private suspend fun getDailyForecastCall(latitude: Double, longitude: Double, millis: Long): CallResult<Forecast> {
+        val response = FrogWeatherApi.dailyForecastRetrofitService.getDailyForecast(latitude, longitude, millis)
+        return if (!response.isSuccessful) {
+            CallResult.Error(Exception(response.errorBody().toString()))
+        } else {
+            if (response.body() == null) {
+                CallResult.Error(Exception("Response body cannot be null."))
+            } else {
+                CallResult.Success(response.body()!!)
```

Comment:
I said before, extract response.body() to variable and after that check for nullability, you will not need to use !! anymore